File: 프로그래머스/42587 프린터/Solution/solution.cpp
#include <string>
#include <vector>
#include <queue>

using namespace std;

int solution(vector<int> priorities, int location) {
    int answer = 0;
    priority_queue<int> pq;
    
    for(int i : priorities){
        pq.push(i);
    }
    
    bool flag = true;
    int cnt = 0;
    while(flag){
        
        for(int i = 0; i < priorities.size(); i++){
            if(priorities[i] == 0) continue;
            if(priorities[i] == pq.top()){
                priorities[i] = 0;
                pq.pop();
                cnt++;
                
                if(i == location){
                    flag = false;
                    answer = cnt;
                    break;
                }
            }
        }
    }
    
    
    return answer;
}

File: 프로그래머스/42587 프린터/Solution/solution.java
```
import java.util.*;

public int solution(int[] priorities, int location) {

	int[] prints = new int[priorities.length];

	List<int[]> list = new ArrayList<>();
	for (int i = 0; i < priorities.length; i++) {
		list.add(new int[] { i + 1, priorities[i] });
	}

	int priority = 0;
	while (!list.isEmpty()) {
		int index = 0;
		boolean flag = false;
		for (int j = index + 1; j < list.size(); j++) {
			if (list.get(index)[1] < list.get(j)[1]) { // 우선순위 비교 중 우선순위가 높은 것이 있다면
				int[] temp = list.remove(index); // 순서를 맨 뒤로
				list.add(temp);
				flag = true;
				break;
			}
		}

		if (!flag) {
			prints[list.remove(index)[0] - 1] = priority + 1;
			priority++;
		}
		index++;
	}

	return prints[location];
}
```
